src/AllArticles.jsx

```diff
@@ -3,27 +3,49 @@ import { useEffect } from "react";
 import { useState } from "react";
 import ArticlesList from "./ArticlesList";
 import "./CSS/search-results-container.css";
+import { capitalise } from "./UTILS/capitalise";
 
 export default function AllArticles({
   searchResultsArticles,
   setSearchResultArticals,
+  searchParams,
+  setSearchParams,
 }) {
   const [viewResults, setViewResults] = useState(false);
+  const topicName = searchParams.get("topic");
+  const [queryError, setQueryError] = useState(false);
 
   useEffect(() => {
-    getArticles()
+    getArticles(topicName)
       .then((response) => {
         setSearchResultArticals(response.articles);
         setViewResults(true);
       })
       .catch((err) => {
         console.log("error: ", err);
+        if (err.response.data.msg === "Bad request: query does not exist") {
+          setQueryError(true);
+        }
       });
-  }, []);
+  }, [topicName]);
+
```

Comment:
Please can you add a loading pattern to your app, it can be very necessary when Render is slow to deliver the backend info ⌛ 
https://notes.northcoders.com/courses/js-front-end/useEffect#loading-patterns